src/Controller/Admin/AdminController.php

```diff
@@ -225,4 +233,70 @@ protected function persistTalkEntity(Talk $talk, Form $newForm)
         $this->em->persist($talk);
         $this->em->flush();
     }
+
+    protected function removeUserEntity(User $user): void
+    {
+        $participations = $this->participationRepository->findBy(['participant' => $user]);
+
+        foreach ($participations as $participation) {
+            $participation->setParticipant(null);
+        }
+
+        $this->em->remove($user);
+        $this->em->flush();
+    }
+
+    protected function removeNextConferenceEntity(Conference $conference): void
+    {
+        $submits = $this->submitRepository->findBy(['conference' => $conference]);
+
+        foreach ($submits as $submit) {
+            $submit->setConference(null);
+        }
+
+        $participations = $this->participationRepository->findBy(['conference' => $conference]);
+
+        foreach ($participations as $participation) {
+            $participation->setConference(null);
+        }
+
+        $this->em->remove($conference);
+        $this->em->flush();
+    }
+
+    protected function removePassedConferenceEntity(Conference $conference): void
+    {
+        $submits = $this->submitRepository->findBy(['conference' => $conference]);
+
+        foreach ($submits as $submit) {
+            $submit->setConference(null);
+        }
+
+        $participations = $this->participationRepository->findBy(['conference' => $conference]);
+
+        foreach ($participations as $participation) {
+            $participation->setConference(null);
+        }
+
+        $this->em->remove($conference);
+        $this->em->flush();
+    }
+
+    protected function removeExcludedConferencesEntity(Conference $conference): void
+    {
+        $submits = $this->submitRepository->findBy(['conference' => $conference]);
+
+        foreach ($submits as $submit) {
+            $submit->setConference(null);
+        }
+
+        $participations = $this->participationRepository->findBy(['conference' => $conference]);
+
+        foreach ($participations as $participation) {
+            $participation->setConference(null);
+        }
+
+        $this->em->remove($conference);
+        $this->em->flush();
```

Comment:
You must centralize the duplicated code in a private method then call it in the 3 remove*ConferencesEntity methods